ngraph/frontend/paddlepaddle/src/op/yolo_box.cpp

```diff
@@ -0,0 +1,262 @@
+    auto input_size = input_height * downsample_ratio;
+
+    std::vector<int64_t> score_shape {1, input_height * input_width * num_anchors, class_num};
+    
+    auto conf_thresh = node_context.get_attribute<float>("conf_thresh");
+    std::vector<float> conf_thresh_mat(score_shape[1], conf_thresh);
+
+    std::cout << "input_height: " << input_height << " input_width: " << input_width << " input_size: " << input_size<< std::endl;
+    std::cout << "num_anchors: " << num_anchors << " scale_x_y: " << scale_x_y << std::endl;
+    std::cout << "downsample_ratio: " << downsample_ratio << " conf_thresh: " << conf_thresh << std::endl;
+    std::cout << "class_num:  " << class_num << " image_size: " << image_size << std::endl;
+
+    auto clip_bbox = node_context.get_attribute<bool>("clip_bbox"); 
+
+    // main X
+    auto node_x_shape = Constant::create<int64_t>(i64, {5}, 
+                                                {1, num_anchors, 5 + class_num, input_height, input_width});
+
+    auto node_x_reshape = std::make_shared<Reshape>(data, node_x_shape, false);
+
+    auto node_input_order = Constant::create(i64, {5}, {0, 1, 3, 4, 2});   
+    auto node_x_transpose = std::make_shared<Transpose>(node_x_reshape, node_input_order); 
+
+    //  range x/y
+    std::vector<float> range_x(input_width);
+    std::iota(range_x.begin(), range_x.end(), 0);
+    std::vector<float> range_y(input_height);
+    std::iota(range_y.begin(), range_y.end(), 0);
+    
+    auto node_range_x = Constant::create<float>(f32, {1, range_x.size()}, range_x);
+    auto node_range_y = Constant::create<float>(f32, {range_y.size(), 1}, range_y);
+    
+    auto node_range_x_shape = Constant::create<int64_t>(i64, {2}, {1, input_width});
+    auto node_range_y_shape = Constant::create<int64_t>(i64, {2}, {input_height, 1});
+                                                 
+    auto node_grid_x = std::make_shared<Tile>(node_range_x, node_range_y_shape);
+    auto node_grid_y = std::make_shared<Tile>(node_range_y, node_range_x_shape);
+
+    // main X (part2)
+    auto node_split_axis = Constant::create<int64_t>(i64, {1}, {-1});
+    auto node_split_lengths = Constant::create<int64_t>(i64, {6}, {1, 1, 1, 1, 1, class_num});
+    auto node_split_input = std::make_shared<VariadicSplit>(node_x_transpose, node_split_axis, node_split_lengths);
+
+    auto node_box_x = node_split_input->output(0);
+    auto node_box_y = node_split_input->output(1);
+    auto node_box_w = node_split_input->output(2);
+    auto node_box_h = node_split_input->output(3);
+    auto node_conf = node_split_input->output(4);
+    auto node_prob = node_split_input->output(5);
+
+    // x/y
+    std::shared_ptr<ngraph::Node> node_box_x_sigmoid = std::make_shared<Sigmoid>(node_box_x);
+    std::shared_ptr<ngraph::Node> node_box_y_sigmoid = std::make_shared<Sigmoid>(node_box_y);
+
+    if (scale_x_y != default_scale) { //FIXME: float compare
+        // TODO
```

Comment:
Please describe what this todo is about

Reply:
@ceciliapang2011 Please have a look

Reply:
fixed. 6c451cefc4bc84c3086f1a8f37517912bbbf0f7c